### PR TITLE
Data Masking Convenience

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
## Context

AI/ai-sdk-java-backlog#102.

As an orchestration user, I want to have a convenient way of configuring data masking.

### Feature scope:
 
- [x] Created `withDpiMaskingConfig(DpiMaskingConfig)`

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] Error handling created / updated & covered by the tests above
- [ ] ~Aligned changes with the JavaScript SDK~
- [x] Documentation updated
- [ ] ~Release notes updated~

### CODE REVIEW

File: docs/guides/ORCHESTRATION_CHAT_COMPLETION.md [hunk 1]
```diff
@@ -184,15 +184,8 @@ var result =
...
-var maskingProvider =
-    MaskingProviderConfig.create()
-        .type(MaskingProviderConfig.TypeEnum.SAP_DATA_PRIVACY_INTEGRATION)
-        .method(MaskingProviderConfig.MethodEnum.ANONYMIZATION)
-        .entities(
-            DPIEntityConfig.create().type(DPIEntities.PHONE),
-            DPIEntityConfig.create().type(DPIEntities.PERSON));
-var maskingConfig = MaskingModuleConfig.create().maskingProviders(maskingProvider);
-var configWithMasking = config.withMaskingConfig(maskingConfig);
+var maskingConfig = DpiMaskingConfig.pseudonymization().withEntities(DPIEntities.PERSON, DPIEntities.EMAIL);
```
Comment (MatKuhr) [inline docs/guides/ORCHESTRATION_CHAT_COMPLETION.md:187]: With this change to pseudo the example prompt doesn't fit anymore

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java [hunk 2]
```diff
@@ -62,4 +62,17 @@ public class OrchestrationModuleConfig {
...
+  /**
+   * Creates a new configuration with the given Data Masking configuration.
+   *
+   * @param dpiMasking The Data Masking configuration to use.
+   * @return A new configuration with the given Data Masking configuration.
+   */
+  @Tolerate
+  @Nonnull
+  public OrchestrationModuleConfig withDpiMaskingConfig(
+      @Nonnull final DpiMaskingConfig dpiMasking) {
```
Comment (MatKuhr) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java:75]: Masking has a similar challenge like filtering: multiple masking provider configs are allowed, and they can be from different providers. 

Here, we always replace the entire masking module config, which is t ideal. Maybe compare with how the filtering PR solves the problem

Comment (CharlesDuboisSAP) [reply orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java:75]: Ok, then I will not call it `with` but `add`

Comment (MatKuhr) [reply orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java:75]: `add` has the problem that it sounds like we are mutating, but our object is designed to be immutable

File: orchestration/src/test/java/com/sap/ai/sdk/orchestration/ConfigToRequestTransformerTest.java [hunk 3]
```diff
@@ -75,4 +78,50 @@ void testMessagesHistory() {
...
+
+    var configModified = config.withMaskingConfig(maskingConfig);
+    assertThat(configModified.getMaskingConfig()).isNotNull();
+    assertThat(configModified.getMaskingConfig().getMaskingProviders())
+        .withFailMessage("withMaskingConfig() should overwrite the existing config and not append")
+        .hasSize(1);
+  }
+
+  @Test
+  void testLLMConfig() {
```
Comment (CharlesDuboisSAP) [inline orchestration/src/test/java/com/sap/ai/sdk/orchestration/ConfigToRequestTransformerTest.java:107]: Drive by test

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    public DpiMasking withEntities(
+        @Nonnull final DPIEntities entity, @Nonnull final DPIEntities... entities) {
+      val entitiesList = new ArrayList<DPIEntities>();
+      entitiesList.add(entity);
+      entitiesList.addAll(Arrays.asList(entities));
+      return new DpiMasking(maskingMethod, entitiesList);
+    }
+  }
+
+  /** {@inheritDoc} */
```
Comment (MatKuhr) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java:78]:
```suggestion
```

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    public DpiMasking withEntities(
+        @Nonnull final DPIEntities entity, @Nonnull final DPIEntities... entities) {
+      val entitiesList = new ArrayList<DPIEntities>();
+      entitiesList.add(entity);
+      entitiesList.addAll(Arrays.asList(entities));
+      return new DpiMasking(maskingMethod, entitiesList);
+    }
+  }
+
+  @Nonnull
```
Comment (MatKuhr) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java:78]:
```suggestion
  @Nonnull
  @Override
```

File: docs/guides/ORCHESTRATION_CHAT_COMPLETION.md [hunk 6]
```diff
@@ -210,7 +204,7 @@ var result =
     new OrchestrationClient().chatCompletion(prompt, configWithMasking);
 ```
 
-In this example, the input will be masked before the call to the LLM. Note that data cannot be unmasked in the LLM output.
+In this example, the input will be masked before the call to the LLM and unmasked in the output.
```
Comment (MatKuhr) [inline docs/guides/ORCHESTRATION_CHAT_COMPLETION.md:207]:
```suggestion
In this example, the input will be masked before the call to the LLM and will remain masked in the output.
```

**Review by MatKuhr — APPROVED** [review]
LGTM

**Pull request merged**